compat/compat_mman.c

```diff
@@ -0,0 +1,279 @@
+/*
```

Comment:
We should at least log with `BT_LOGF()` before the `abort()`s below to know what went wrong exactly if that ever happens. See `doc/logging-guide.adoc` to learn how to properly instrument a module. I guess the environment variable would be `BABELTRACE_COMPAT_LOG_LEVEL` and the tags can be `COMPAT-MMAN` and `COMPAT-UUID`.